### PR TITLE
Remove use of BlockStore from Validate.sequenceNumber

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

We can make use of the blockMeta's seqNum instead

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

https://rchain.atlassian.net/browse/RCHAIN-3046

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by KentShikama** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/Validate.scala [hunk 1]
```diff
@@ -342,17 +342,22 @@ object Validate {
...
-                                            latestBlock <- ProtoUtil
-                                                            .unsafeGetBlock[F](latestBlockHash)
-                                          } yield latestBlock.seqNum
+                                          dag.lookup(latestBlockHash).map {
+                                            case Some(block) =>
+                                              block.seqNum
+                                            case None =>
+                                              throw new Exception(
+                                                s"Latest block hash ${PrettyPrinter.buildString(latestBlockHash)} is missing from block dag store."
+                                              )
```
Comment (itegulov) [inline casper/src/main/scala/coop/rchain/casper/Validate.scala]: Why not use `Sync[F].raiseError`?

Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/Validate.scala]: I'll do this when I thread through Sync into all the Validate methods.

Comment (itegulov) [reply casper/src/main/scala/coop/rchain/casper/Validate.scala]: Ok, I will trust you on this

**Pull request merged**